src/pyhf/infer/intervals.py

```diff
@@ -2,13 +2,96 @@
 from . import hypotest
 from .. import get_backend
 import numpy as np
+from scipy.optimize import toms748 as _toms748
 
 
 def _interp(x, xp, fp):
     tb, _ = get_backend()
     return tb.astensor(np.interp(x, xp.tolist(), fp.tolist()))
 
 
+def upperlimit_auto(data, model, low, high, level=0.05, atol=2e-12, rtol=1e-15):
```

Comment:
This is personal taste, but I think the default tolerance (I think it is tolerance in mu and not the CLs, but could be wrong) could be much larger, e.g. `0.01`.

Reply:
I've set the default tolerances to the minimum possible. It might make sense to have no default relative tolerance, and make it a required parameter instead.

Reply:
OK, we now emit a warning if the user doesn't provide an `rtol` explaining that it should be set as high as is acceptable.